src/solver/solve_helpers.jl

```diff
@@ -0,0 +1,201 @@
+# general helpers
+function ssubs(expr, dict)
+    if haskey(dict, expr)
+        return dict[expr]
+    end
+    expr = unwrap(expr)
+    if iscall(expr)
+        op = ssubs(operation(expr), dict)
+        args = map(arguments(expr)) do x
+            ssubs(x, dict)
+        end
+        op(wrap.(args)...)
+    else
+        return expr
+    end
+end
+
+function ssqrt(n)
+    n = unwrap(n)
+
+    if n isa Real
+        n > 0 && return sqrt(n)
```

Comment:
@n0rbed if you'd have time, can you put a small comment here to explain why we use our own `ssqrt` instead of the `Base.sqrt` ?Something like
> In the solver we might want to return `Symbolics.term(sqrt, x)` as a root. If `x` is a negative real number, the evaluation of such a root will fail since `sqrt` is not defined. Thus we use `Symbolics.term(Symbolics.ssqrt, x)` that handles negative numbers.